### PR TITLE
Api team

### PR DESCRIPTION
**Pull request by chriscastaneda**

### CODE REVIEW

**Comment by mgoshorn** [comment]
Please fix reported vulnerabilities and code smells.  Most of them should be fairly trivial fixes.

**Comment by mgoshorn** [comment]
Looks like there are still some issues to be fixed. To fix all issues:

- [ ] Remove all tests which do not make any assertions.  We should not have tests that do not do anything.

- [ ] Replace usage of hibernate entities at the controller level with DTO objects that we programmatically transform into DTOs.  See SonarCloud for details on why not doing this introduces security vulnerabilities.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,7 @@
 /target
 /RemoteSystemsTempFiles
 /bin
+/src/main/resources/application.yaml
```
Comment (mgoshorn) [inline .gitignore]: Why are we ignoring configuration files?

File: src/test/java/com/revature/controllers/QuestionControllerTests.java [hunk 2]
```diff
@@ -60,5 +60,49 @@ public void testGetAllQuestionsHappyPath() throws Exception {
...
+		mvc.perform(get("/questions/user/1")
+			.accept(MediaType.APPLICATION_JSON))
+			.andExpect(status().isOk())
+			.andExpect(content()
+					.contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+			.andExpect(jsonPath("$.content[0].id", is(1)));
+			
+	}
+	
+	/* @Author ken
```
Comment (mgoshorn) [inline src/test/java/com/revature/controllers/QuestionControllerTests.java]: Why is this test commented out? We could use @Ignore to not run the test, but if the test has no value then it should be removed. If it is failing, then the feature should be fixed such that the test passes.

File: src/test/java/com/revature/controllers/test/AnswerControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+				.accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk())
+				.andExpect(content()
+						.contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+				.andExpect(jsonPath("$.content[0].id", is(1)));
+	}
+	
+	/*
+	@Test
+	public void testSaveAnswer() throws Exception {
```
Comment (mgoshorn) [inline src/test/java/com/revature/controllers/test/AnswerControllerTest.java:75]: Same as other commented out test:
Why is this test commented out? We could use @Ignore to not run the test, but if the test has no value then it should be removed. If it is failing, then the feature should be fixed such that the test passes.

File: src/test/java/com/revature/services/test/QuestionService.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.revature.services.test;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class QuestionService {
```
Comment (mgoshorn) [inline src/test/java/com/revature/services/test/QuestionService.java]: This test file provides no actual testing. Remove it.

File: src/test/java/com/revature/services/test/UserServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.revature.services.test;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class UserServiceTest {
```
Comment (mgoshorn) [inline src/test/java/com/revature/services/test/UserServiceTest.java:8]: This test file provides no actual testing. Remove it.

**Review by mgoshorn — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (kRenelt) [reply .gitignore]: fixed

File: src/test/java/com/revature/controllers/QuestionControllerTests.java (hunk 2 above)
Comment (kRenelt) [reply src/test/java/com/revature/controllers/QuestionControllerTests.java]: fixed

File: src/test/java/com/revature/controllers/test/AnswerControllerTest.java (hunk 3 above)
Comment (kRenelt) [reply src/test/java/com/revature/controllers/test/AnswerControllerTest.java:75]: code has been added and uncommented

File: src/test/java/com/revature/services/test/QuestionService.java (hunk 4 above)
Comment (kRenelt) [reply src/test/java/com/revature/services/test/QuestionService.java]: code added

**Pull request closed without merging**